src/libcore/cell.rs

```diff
@@ -280,6 +281,19 @@ unsafe impl<T> Send for Cell<T> where T: Send {}
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T> !Sync for Cell<T> {}
 
+#[unstable(feature = "move_cell", issue = "39264")]
+impl<T:Clone> Clone for Cell<T> {
+    #[inline]
+    default fn clone(&self) -> Cell<T> {
+        let temp = unsafe { mem::uninitialized() };
+        let inner = self.replace(temp);
+        // If `clone` panics, let it crash.
```

Comment:
Isn't this super unsafe? If a panic occurs here, `*self` will be dropped which will drop uninitialized memory which will cause UB.